--- src/Tools.cpp	2018-01-12 12:12:08.254349300 +0100
+++ New_Tools.cpp	2018-01-12 19:21:51.956924100 +0100
@@ -201,8 +201,12 @@
           default: break;
           }
         }
+#if defined(_MSC_VER) && _MSC_VER < 1900
+        _snprintf(buffer,1024,"%s%s%s_%s%s",
+#else
         std::snprintf(buffer,1024,"%s%s%s_%s%s",
-                      paths[i],separator,prefix,suffix,extension);
+#endif
+      paths[i],separator,prefix,suffix,extension);
       } while ( canReadFile(buffer) );
       if ( canCreateFile(buffer) ) {
         path = paths[i];
